File: bus_client/lib/models/schedule.dart
class Schedule {
  final int id;
  final int routeId;
  final String routeName;
  final String departureTime;
  final String dayType;

  const Schedule({
    required this.id,
    required this.routeId,
    required this.routeName,
    required this.departureTime,
    required this.dayType,
  });

  factory Schedule.fromJson(Map<String, dynamic> json) {
    return Schedule(
      id: json['id'] as int,
      routeId: json['route_id'] as int? ?? 0,
      routeName: json['route_name'] as String? ?? '',
      departureTime: json['departure_time'] as String,
      dayType: json['day_type'] as String? ?? 'weekday',
    );
  }
}

File: bus_server/app/models/schedule.py
from sqlalchemy import Boolean, Integer, String
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base


class Schedule(Base):
    __tablename__ = "schedules"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    route_id: Mapped[int] = mapped_column(Integer, index=True)
    departure_time: Mapped[str] = mapped_column(String(5))  # "HH:MM" format
    day_type: Mapped[str] = mapped_column(String(20), default="weekday")  # weekday, saturday, holiday
    is_active: Mapped[bool] = mapped_column(Boolean, default=True)
